File: sem2/modele_systemow_dynamicznych/lista1/preyPredator.m
function preyPredator(xzero, yzero, params, T, dt)
    t = 0:dt:T;
    xprev = xzero;
    yprev = yzero;
    a = params(1);
    b = params(2);
    c = params(3);
    d = params(4);
    xvec = zeros(size(t));
    yvec = zeros(size(t));

    for iter = 1:length(t)
        dxdt = (a - b * yprev) * xprev;
        dydt = (c * xprev - d) * yprev;

        xcurr = xprev + dxdt * dt;
        ycurr = yprev + dydt * dt;

        xvec(iter) = xcurr;
        yvec(iter) = ycurr;

        xprev = xcurr;
        yprev = ycurr;
    end
    plot(t, xvec, "blue", t, yvec, "red", T, dt);
%    legend("Prey population", "Predator population");
end

File: sem2/modele_systemow_dynamicznych/lista3/prey_predator.py
from numpy import arange
import matplotlib.pyplot as plt

plotLineWidth = 0.8

def preyPredatorEuler(xZero, yZero, params, t, dt):
    timeList = [round(x, 3) for x in arange(0, t, dt)]
    xPrev = xZero
    yPrev = yZero
    a = params[0]
    b = params[1]
    c = params[2]
    d = params[3]
    xList = [0]*len(timeList)
    yList = [0]*len(timeList)

    for i in range(0, len(xList)):
        dxdt = (a - b * yPrev) * xPrev
        dydt = (c * xPrev - d) * yPrev

        xList[i] = xPrev + dxdt * dt
        yList[i] = yPrev + dydt * dt

        xPrev = xList[i]
        yPrev = yList[i]

    

    plt.plot(timeList, xList, 'r', timeList, yList, 'b', lw=plotLineWidth)
    plt.legend(['X', 'Y'])
    plt.show()
